From 873972aaf59e9cdc3ffaddf0e23d6a13d35a59da Mon Sep 17 00:00:00 2001
From: David Edmundson <davidedmundson@kde.org>
Date: Fri, 10 Oct 2025 11:16:53 +0100
Subject: [PATCH 373/553] wayland: Fix subsurface buffers outliving the backing
 store queue

In the path to handle subsurfaces that are painted as part of another
window, QWaylandShmBacking store may create temporary buffers for the
subsurface window.

This buffer has it's own lifespan managed by autodeletion from the
compositor not from the backing store. As it can outlive the
QWaylandShmBackingStore it should not share the same event queue as
having an object outlive the event queue can be problematic. As we
create a new buffer every time the custom event queue is not needed
here.

Amends: a0161a2b783c1cbea8f1faaa6fe9f14160515900

Change-Id: I7815cf8c9fc539e739499a9e637b1e52cb9c92f3
Reviewed-by: Vlad Zahorodnii <vlad.zahorodnii@kde.org>
(cherry picked from commit aa684238c5285bc5ae7741e79f77acc9726d38b7)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/plugins/platforms/wayland/qwaylandshmbackingstore.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/platforms/wayland/qwaylandshmbackingstore.cpp b/src/plugins/platforms/wayland/qwaylandshmbackingstore.cpp
index 9a67190595c..d525fe18b8a 100644
--- a/src/plugins/platforms/wayland/qwaylandshmbackingstore.cpp
+++ b/src/plugins/platforms/wayland/qwaylandshmbackingstore.cpp
@@ -270,7 +270,7 @@ void QWaylandShmBackingStore::flush(QWindow *window, const QRegion &region, cons
         const auto scale = waylandWindow->scale();
         auto newBuffer = new QWaylandShmBuffer(
                 mDisplay, window->size() * scale, mBackBuffer->image()->format(),
-                mBackBuffer->image()->devicePixelRatio(), mEventQueue);
+                mBackBuffer->image()->devicePixelRatio());
         newBuffer->setDeleteOnRelease(true);
         QRect sourceRect(offset * scale, window->size() * scale);
         QPainter painter(newBuffer->image());
-- 
2.51.2

